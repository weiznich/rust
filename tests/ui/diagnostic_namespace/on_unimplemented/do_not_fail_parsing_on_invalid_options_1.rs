#![feature(diagnostic_namespace, diagnostic_on_unimplemented)]
// check-fail

#[diagnostic::on_unimplemented(unsupported = "foo")]
//~^WARN Malformed on_unimplemented attribut
//~|WARN Malformed on_unimplemented attribut
trait Foo {}

#[diagnostic::on_unimplemented(message = "Baz")]
// todo: generate a warning here as well
struct Bar {}

#[diagnostic::on_unimplemented(message = "Boom", unsupported = "Bar")]
//~^WARN Malformed on_unimplemented attribut
//~|WARN Malformed on_unimplemented attribut
trait Baz {}

#[diagnostic::on_unimplemented(message = "Boom", on(_Self = "i32", message = "whatever"))]
//~^WARN Malformed on_unimplemented attribut
//~|WARN Malformed on_unimplemented attribut
trait Boom {}

#[diagnostic::on_unimplemented = "boom"]
//~^WARN Malformed on_unimplemented attribut
trait Doom {}

fn take_foo(_: impl Foo) {}
fn take_baz(_: impl Baz) {}
fn take_boom(_: impl Boom) {}

fn main() {
    take_foo(1_i32);
    //~^ERROR the trait bound `i32: Foo` is not satisfied
    take_baz(1_i32);
    //~^ERROR Boom
    take_boom(1_i32);
    //~^ERROR Boom
}
