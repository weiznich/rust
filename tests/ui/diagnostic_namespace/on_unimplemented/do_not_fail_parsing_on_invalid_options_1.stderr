warning: Malformed on_unimplemented attribute
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:4:32
   |
LL | #[diagnostic::on_unimplemented(unsupported = "foo")]
   |                                ^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(malformed_diagnostic_attributes)]` on by default

warning: Malformed on_unimplemented attribute
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:13:50
   |
LL | #[diagnostic::on_unimplemented(message = "Boom", unsupported = "Bar")]
   |                                                  ^^^^^^^^^^^^^^^^^^^

warning: Malformed on_unimplemented attribute
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:18:50
   |
LL | #[diagnostic::on_unimplemented(message = "Boom", on(_Self = "i32", message = "whatever"))]
   |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: Malformed on_unimplemented attribute
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:23:1
   |
LL | #[diagnostic::on_unimplemented = "boom"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: Malformed on_unimplemented attribute
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:4:32
   |
LL | #[diagnostic::on_unimplemented(unsupported = "foo")]
   |                                ^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `i32: Foo` is not satisfied
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:32:14
   |
LL |     take_foo(1_i32);
   |     -------- ^^^^^ the trait `Foo` is not implemented for `i32`
   |     |
   |     required by a bound introduced by this call
   |
note: required by a bound in `take_foo`
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:27:21
   |
LL | fn take_foo(_: impl Foo) {}
   |                     ^^^ required by this bound in `take_foo`

warning: Malformed on_unimplemented attribute
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:13:50
   |
LL | #[diagnostic::on_unimplemented(message = "Boom", unsupported = "Bar")]
   |                                                  ^^^^^^^^^^^^^^^^^^^

error[E0277]: Boom
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:34:14
   |
LL |     take_baz(1_i32);
   |     -------- ^^^^^ the trait `Baz` is not implemented for `i32`
   |     |
   |     required by a bound introduced by this call
   |
note: required by a bound in `take_baz`
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:28:21
   |
LL | fn take_baz(_: impl Baz) {}
   |                     ^^^ required by this bound in `take_baz`

warning: Malformed on_unimplemented attribute
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:18:50
   |
LL | #[diagnostic::on_unimplemented(message = "Boom", on(_Self = "i32", message = "whatever"))]
   |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: Boom
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:36:15
   |
LL |     take_boom(1_i32);
   |     --------- ^^^^^ the trait `Boom` is not implemented for `i32`
   |     |
   |     required by a bound introduced by this call
   |
note: required by a bound in `take_boom`
  --> $DIR/do_not_fail_parsing_on_invalid_options_1.rs:29:22
   |
LL | fn take_boom(_: impl Boom) {}
   |                      ^^^^ required by this bound in `take_boom`

error: aborting due to 3 previous errors; 7 warnings emitted

For more information about this error, try `rustc --explain E0277`.
